Library/Homebrew/test/build_environment_spec.rb

```diff
@@ -0,0 +1,70 @@
+require "build_environment"
+
+RSpec::Matchers.define :use_userpaths do
+  match do |actual|
+    actual.userpaths?
+  end
+end
+
+describe BuildEnvironment do
+  let(:env) { described_class.new }
+
+  describe "#<<" do
+    it "returns itself" do
+      expect(env << :foo).to eq env
+    end
+  end
+
+  describe "#merge" do
+    it "returns itself" do
+      expect(env.merge([])).to eq env
+    end
+  end
+
+  describe "#std?" do
+    it "returns true if the environment contains :std" do
+      expect(env << :std).to be_std
```

Comment:
Could this and similar tests be split up into multiple lines? I don't think mutation/setup should happen _inside_ an expectation.

These are specs — we want them to be as simple as possible to read and understand.